tests/session_test.py

```diff
@@ -0,0 +1,140 @@
+import json
+import random
+import unittest
+import sys
+import subprocess
+import time
+
+from hamcrest import assert_that, less_than, greater_than
+from hamcrest import is_
+import pkg_resources
+import requests
+from tests import available_models
+from functools import partial
+
+
+class SessionTest(unittest.TestCase):
+
+    def test_creating_commit_deleting_session_works(self):
+        with NetmanTestApp() as partial_client:
+            client = partial_client(_switch_of_model("cisco"))
+
+            result = _create_session(client, "i_love_sessions")
+            session_id = result.json()['session_id']
+
+            result = client.post("/switches-sessions/" + session_id + "/actions", data='commit')
+            assert_that(result.status_code, is_(204), result.text)
+
+            result = client.delete("/switches-sessions/" + session_id)
+            assert_that(result.status_code, is_(204), result.text)
+
+    def test_sessions_timeout_let_the_next_session_takeover(self):
+        with NetmanTestApp() as partial_client:
+            client = partial_client(_switch_of_model("brocade"))
+
+            start_time = time.time()
+
+            result = _create_session(client, "i_love_sessions")
+            assert_that(time.time() - start_time, is_(less_than(1)))
+
+            first_session_id = result.json()['session_id']
+
+            result = _create_session(client, "i_really_love_sessions")
+            assert_that(time.time() - start_time, is_(greater_than(1)))
+
+            second_session_id = result.json()['session_id']
+
+            result = client.post("/switches-sessions/" + first_session_id + "/actions", data='commit')
+            assert_that(result.status_code, is_(404), 'A session shall fail after 30s')
```

Comment:
Is this supposed to be 2s given the session inactivity timeout set below? How is the inactivity timeout tested, considering nothing "sleeps" for the session inactivity timeout value.